site_scons/buildutils.py

```diff
@@ -472,137 +591,105 @@ def add_RegressionTest(env):
         action=env.Action(regression_test, regression_test_message))
 
 
-def quoted(s):
-    """ Returns the given string wrapped in double quotes."""
-    return '"%s"' % s
-
-
 def mglob(env, subdir, *args):
     """
-    Each arg in args is assumed to be file extension,
-    unless the arg starts with a '^', in which case the remainder
-    of the arg is taken to be a complete pattern.
+    Each argument in ``args`` is assumed to be file extension,
+    unless the arg starts with a ``'^'``, in which case the remainder
+    of the argument is taken to be a complete pattern.
     """
     matches = []
     for ext in args:
-        if ext.startswith('^'):
-            matches += env.Glob(pjoin(subdir, ext[1:]))
+        if ext.startswith("^"):
+            matches += env.Glob(Path(subdir).joinpath(ext[1:]))
         else:
-            matches += env.Glob(pjoin(subdir, '*.%s' % ext))
+            matches += env.Glob(Path(subdir).joinpath(f"*.{ext}"))
     return matches
 
 
-def psplit(s):
-    """
-    Split a path given as a string into a list.
-    This is the inverse of os.path.join.
-    """
-    s = s.strip('/\\')
-    head, tail = os.path.split(s)
-    path = [tail]
-    while head:
-        head, tail = os.path.split(head)
-        path.append(tail)
-
-    path.reverse()
-    return path
-
-
-def subdirs(path):
-    """ Get the subdirectories of a specified directory """
-    for subdir in os.listdir(path):
-        dirpath = pjoin(path, subdir)
-        if os.path.isdir(dirpath):
-            yield subdir
-
-def stripDrive(s):
-    """
-    Remove a Windows drive letter specification from a path.
-    """
-    if len(s) > 1 and s[1] == ':':
-        return s[2:]
-    else:
-        return s
-
-
-def which(program):
-    """ Replicates the functionality of the 'which' shell command """
-    def is_exe(fpath):
-        for ext in ('', '.exe', '.bat'):
-            if os.path.exists(fpath + ext):
-                return os.access(fpath + ext, os.X_OK)
-        return False
-
-    fpath, fname = os.path.split(program)
-    if fpath:
-        if is_exe(program):
-            return program
-    else:
+def which(program: str) -> bool:
+    """Replicates the functionality of the 'which' shell command."""
+    for ext in ("", ".exe", ".bat"):
+        fpath = Path(program + ext)
         for path in os.environ["PATH"].split(os.pathsep):
-            exe_file = os.path.join(path, program)
-            if is_exe(exe_file):
-                return exe_file
-
-    return None
+            exe_file = Path(path).joinpath(fpath)
+            if exe_file.exists() and os.access(exe_file, os.X_OK):
+                return True
+    return False
 
-optionWrapper = textwrap.TextWrapper(initial_indent='    ',
-                                   subsequent_indent='    ',
-                                   width=72)
 
+def help(env, options):
```

Comment:
PEP 484. Docstring is not clear: what options are there?

Reply:
The options are defined in `SConstruct`. They are the build options, not options for this function.

Reply:
that was clear :wink:, but PEP 484 by itself clarifies things here. (if this weren't an interface only seen by people digging deep under the hood, I would recommend to italicize *options* in the docstring)